tox.ini

```diff
@@ -36,4 +36,10 @@ addopts = --ignore=test/manual
 markers = 
 	drivertest : Fixate driver test
 
-
+[testenv:mypy]
+basepython = python3
+deps = mypy==1.2
+# mypy gives different results if you actually install the stuff before you check it
+# don't cache anything, will be slower, but each run should be clean
+# this also stops weirdness around sharing cache with other instances of mypy
+commands = mypy --no-incremental --config-file=mypy.ini
```

Comment:
With `--no-incremental` it takes 15 seconds, with cache it takes 2 seconds (but still takes the same amount of time the first time it is run)
If we don't specify `--no-incremental` we at least will want something like `--cache-dir="{envdir}/mypy_cache"` because I was getting really weird errors due to tox's mypy and my local mypy sharing caches

Reply:
mypy is at version 1.3 now, is there any reason not to use the latest over the currently pinned 1.2?

Reply:
No idea why 1.2 was installed over 1.3, will upgrade that.

I think as well this also only installs the dependencies for the base fixate package, I think I should add a run for the gui as well.

Reply:
Gui has not much in it and looks like there are things in there that don't have type stubs. There's a package that addresses some of this PyQt5-stubs. I'll leave everything for now and look into it if we have to do any changes to the gui in the future.

Reply:
> The main thing I'm wary off is not having a clear goal stated. For example, that goal could be
>  - Have all interfaces intended for use in scripts annotated

> And I've already said this before, but not sure I've committed it in writing - we almost certainly shouldn't try annotating the code as is. There is all kinds of busted stuff that most likely needs to be refactored and/or redesigned before annotation will be useful.

After looking at this again I think this is the way to go, the current mypy config was written using default settings and turn off stuff that breaks, with the intent of fixing things up and turning them back on. The only problem is that as we turn on stricter and stricter settings this will cause all sorts of things to pop up.

I think the main priority is to make things easier to use in test scripts.

I'd say the things that would making writing test scripts easiest would be
- drivers
- jig_mapping (needs virtualmux rework)
- checks (John added some docstrings explaining what the checks actually do now, typing these would really only enforce whether these are super generic comparisons or meant for numeric types only)
- ui (not cli/qt specific just the generic interface)


Some things that would hopefully be easy to add in
- common (other than testclasses I had no idea all the other stuff was in here)
- control
- exceptions (really need to change these from BaseException at some point)


Modules that could benefit at some point:
- reporting

Maybe other modules can be enabled partially/fully if people work on them?
A lot of things are hard to work on because unstructured clumps of data get passed between modules, in a lot of modules current states, mypy would probably complain a lot